src/crypto/CHIPCryptoPAL.h

```diff
@@ -99,15 +99,45 @@ enum class SupportedECPKeyTypes : uint8_t
     ECP256R1 = 0,
 };
 
+template <typename Sig>
 class ECPKey
 {
 public:
     virtual SupportedECPKeyTypes Type() const = 0;
     virtual size_t Length() const             = 0;
     virtual operator uint8_t *() const        = 0;
+
+    virtual CHIP_ERROR ECDSA_validate_msg_signature(const uint8_t * msg, const size_t msg_length, const Sig & signature) const
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+};
+
+class P256ECDSASignature
+{
+public:
+    size_t & Length() { return length; }
```

Comment:
this permits a "set()" operation on length?

Reply:
Yes

Reply:
Why can't we do something more obvious like SetLength() ?

Reply:
+1 to this ... I would find 

foo.Length() = 123;

to be more alien than

foo.SetLength(123);

I would that if length needs to be passed in as some argument, this could be somehow needed, but then maybe instead of

```
Process(data, foo.Length());
```

it would be clearer to do

```
size_t len = foo.Length();
Process(data, len);
foo.SetLength(len);
```